### PR TITLE
[SPARK-26547][SQL] Remove duplicate toHiveString from HiveUtils

### PR DESCRIPTION
**Pull request by MaxGekk**
## What changes were proposed in this pull request?

The `toHiveString()` and `toHiveStructString` methods were removed from `HiveUtils` because they have been already implemented in `HiveResult`. One related test was moved to `HiveResultSuite`. 

## How was this patch tested?

By tests from `hive-thriftserver`.

### CODE REVIEW

**Comment by MaxGekk** [comment]
@hvanhovell Please, take a look at the changes.

**Comment by MaxGekk** [comment]
jenkins, retest this, please

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala [hunk 1]
```diff
@@ -444,52 +444,6 @@ private[spark] object HiveUtils extends Logging {
...
-    case (null, _) => "NULL"
-    case (d: Int, DateType) => new DateWritable(d).toString
-    case (t: Timestamp, TimestampType) => new TimestampWritable(t).toString
-    case (bin: Array[Byte], BinaryType) => new String(bin, StandardCharsets.UTF_8)
-    case (decimal: java.math.BigDecimal, DecimalType()) =>
-      // Hive strips trailing zeros so use its toString
-      HiveDecimal.create(decimal).toString
-    case (other, _ : UserDefinedType[_]) => other.toString
-    case (other, tpe) if primitiveTypes contains tpe => other.toString
-  }
```
[the remark this answers is not in the document]
Comment (MaxGekk) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala]: done

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @MaxGekk .

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Pull request closed without merging**